integration_tests/eip712_utils.py

```diff
@@ -0,0 +1,415 @@
+import ethermint.types.v1.web3_pb2
+import os
+from pathlib import Path
+from dotenv import load_dotenv
+from eth_account import Account
+from eth_account.messages import encode_structured_data
+
+MSG_SEND_TYPES = {
+    "MsgValue": [
+      { "name": "from_address", "type": "string" },
+      { "name": "to_address", "type": "string" },
+      { "name": "amount", "type": "TypeAmount[]" },
+    ],
+    "TypeAmount": [
+      { "name": "denom", "type": "string" },
+      { "name": "amount", "type": "string" },
+    ],
+}
+
+LEGACY_AMINO = 127
+SIGN_DIRECT = 1
+
+load_dotenv(Path(__file__).parent.parent / "scripts/.env")
+Account.enable_unaudited_hdwallet_features()
+ACCOUNTS = {
+    "community": Account.from_mnemonic(os.getenv("COMMUNITY_MNEMONIC")),
+}
+KEYS = {name: account.key for name, account in ACCOUNTS.items()}
+
+def create_message_send(chain, sender, fee, memo, params):
+    # EIP712
+    fee_object = generate_fee(
+        fee["amount"],
+        fee["denom"],
+        fee["gas"],
+        sender["accountAddress"],
+    )
+    types = generate_types(MSG_SEND_TYPES)
+    msg = create_msg_send(
+        params["amount"],
+        params["denom"],
+        sender["accountAddress"],
+        params["destinationAddress"],
+    )
+    messages = generate_message(
+        str(sender["accountNumber"]),
+        str(sender["sequence"]),
+        chain["cosmosChainId"],
+        memo,
+        fee_object,
+        msg,
+    )
+    eip_to_sign = create_eip712(types, chain["chainId"], messages)
+    msg_send = proto_msg_send(
+        sender["accountAddress"],
+        params["destinationAddress"],
+        params["amount"],
+        params["denom"],
+    )
+    tx = create_transaction(
+        msg_send,
+        memo,
+        fee["amount"],
+        fee["denom"],
+        fee["gas"],
+        "ethsecp256",
+        sender["pubkey"],
+        sender["sequence"],
+        sender["accountNumber"],
+        chain["cosmosChainId"],
+    )
+    return {
+        "signDirect": tx["signDirect"],
+        "legacyAmino": tx["legacyAmino"],
+        "eipToSign": eip_to_sign,
+    }
+
+
+def generate_fee(amount, denom, gas, fee_payer):
+    return {
+        "amount": [{
+            "amount": amount,
+            "denom": denom,
+        }],
+        "gas": gas,
+        "feePayer": fee_payer,
+    }
+
+
+def generate_types(msg_values):
+    types = {
+        "EIP712Domain": [
+            { "name": "name", "type": "string" },
+            { "name": "version", "type": "string" },
+            { "name": "chainId", "type": "uint256" },
+            { "name": "verifyingContract", "type": "string" },
+            { "name": "salt", "type": "string" },
+        ],
+        "Tx": [
+            { "name": "account_number", "type": "string" },
+            { "name": "chain_id", "type": "string" },
+            { "name": "fee", "type": "Fee" },
+            { "name": "memo", "type": "string" },
+            { "name": "msgs", "type": "Msg[]" },
```

Comment:
@yihuang we miss regex support for array for `Msg[]` and `Coin[]` in python [lib](https://github.com/ethereum/eth-account/blob/41f56e45b49ec1966a72cba12955bc1185cf7284/eth_account/_utils/structured_data/hashing.py#L262), since [metamask](https://github.com/MetaMask/eth-sig-util/blob/e293ea9f071a4605205746e97691370229ebc227/src/sign-typed-data.ts#L287) extract all character

Reply:
[fix](https://github.com/ethereum/eth-account/pull/175/files#diff-658f7167e093f1696dcd657c9bd1b01757bb8933dd27d3d366904acdcfe18762) in latest master but they hasn't release in [v0.6.1](https://github.com/ethereum/eth-account/blob/v0.6.1/eth_account/_utils/structured_data/hashing.py) yet